### PR TITLE
[T5A2][T15-A3]Kenny Ng

### PR DESCRIPTION
**Pull request by kennyngdsc**
Work in progress

### CODE REVIEW

**Comment by kennyngdsc** [comment]
Ready for review

File: src/seedu/addressbook/commands/AddCommand.java [hunk 1]
```diff
@@ -58,13 +58,17 @@ public ReadOnlyPerson getPerson() {
     }
 
     @Override
-    public CommandResult execute() {
+    public CommandResult execute() /* throws Exception */ {
```
Comment (sooyj) [inline src/seedu/addressbook/commands/AddCommand.java:61]: This portion is incomplete. Have you figured out why there is compilation error?

File: src/seedu/addressbook/commands/AddCommand.java [hunk 2]
```diff
@@ -58,13 +58,17 @@ public ReadOnlyPerson getPerson() {
...
         try {
             addressBook.addPerson(toAdd);
             return new CommandResult(String.format(MESSAGE_SUCCESS, toAdd));
         } catch (UniquePersonList.DuplicatePersonException dpe) {
             return new CommandResult(MESSAGE_DUPLICATE_PERSON);
         }
     }
+    
+    public boolean isMutating() {
```
Comment (sooyj) [inline src/seedu/addressbook/commands/AddCommand.java:70]: Correct example of a mutating command.

File: src/seedu/addressbook/commands/Command.java [hunk 3]
```diff
@@ -65,4 +65,9 @@ public int getTargetIndex() {
     public void setTargetIndex(int targetIndex) {
         this.targetIndex = targetIndex;
     }
+    
+    /**
+     * Returns true for command types that mutate data
+     */
+    public abstract boolean isMutating();
```
Comment (sooyj) [inline src/seedu/addressbook/commands/Command.java:72]: Correct abstract method. Make sure you understand the reason for the "abstract" keyword here.

File: src/seedu/addressbook/commands/ExitCommand.java [hunk 4]
```diff
@@ -16,4 +16,7 @@ public CommandResult execute() {
         return new CommandResult(MESSAGE_EXIT_ACKNOWEDGEMENT);
     }
 
+    public boolean isMutating() {
```
Comment (sooyj) [inline src/seedu/addressbook/commands/ExitCommand.java:19]: Correct example of a non-mutating command.

File: src/seedu/addressbook/logic/Logic.java [hunk 5]
```diff
@@ -85,7 +85,9 @@ public CommandResult execute(String userCommandText) throws Exception {
     private CommandResult execute(Command command) throws Exception {
         command.setData(addressBook, lastShownList);
         CommandResult result = command.execute();
-        storage.save(addressBook);
+        if (command.isMutating()) {
```
Comment (sooyj) [inline src/seedu/addressbook/logic/Logic.java:88]: Due to the new behavior of the "save" method, some of the tests may be broken. Think about how to resolve the tests failures.

**Review by sooyj — COMMENTED** [review]
In complete attempt. The exercise on the LSP is not entirely done.

Please approach me if you still have issue understanding the LSP portion.

**Pull request closed without merging**